### ISSUE TITLE
Handling of external drives

### ISSUE THREAD

**Issue by raev33** [duplicati/duplicati#2768]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info
- **Duplicati version**: 2.0.2.1
- **Operating system**: Ubuntu 16.04
- **Backend**: Backblaze B2

## Description
I have a backup with two main directories. One local and one on an external hard drive. The files on the local hard drive are files i'm using right now - and from time to time I archive them on the external hard drive - this means there are a lot of identical files in the two directories.
The issue is that Duplicati only can start a backup when the external hard drive is plugged in, otherwise it simply dismisses the task - though it still can access the local main directory.

## Steps to reproduce
1. Make a new backup with a local directory and a directory on an external hard drive.
2. Unplug the external hard drive.
3. Try to start the backup.

- **Actual result**:
Duplicati dismisses the whole backup process.
- **Expected result**:
Duplicati should backup the local directory, even though the external directory is missing.
It would be even better if there was an option to mark external drives - so that a missing main directory, not necessarily would be read as a deleted directory

**Comment by Pectojin** [comment]
I believe this is issue is the same as my issue duplicati/duplicati#2760 or at least a fix would likely resolve both.

Also, interesting point about external drives then being considered "deleted". There isn't really be a way to know for sure if it was deleted or removed until Duplicati gets active continious backup that monitors changes on disk. 

If you need to recover those "not deleted files" you could go back to a previous backup to restore the external files, but if you only have the drive connected occassionally you'd have to look around for the right backup job, which contains the files.

Crashplan's restore view is a bit more useful for that kind of situation, since it can show all files including deleted and then lets you pick a version for the specific file/folder instead of viewing a given restore point.

**Comment by kenkendk** [comment]
@raev33 You can add the option `--allow-missing-source` if you want to ignore the missing source drive. Does this solve your problem?

**Comment by raev33** [comment]
Yes, partially. I can now backup the local data, but then my external data is still deleted on the backup.

**Comment by alexgreen** [comment]
Any further updates in this? I'm considering using Duplicati but the inability to effectively handle external drives that aren't continuously connected would be a total deal-breaker for me.

**Comment by Pectojin** [comment]
I'm not aware of any progress having been made specifically on external drives. There has been some work on using USN to keep track of file changes on Windows, but that doesn't immediately translate into a solution for this.

**Comment by stewartadam** [comment]
+1 was hoping to setup Duplicati to backup local files on the PC as well as files from an external drive when attached, would be nice to have an option skip, not erase, backups with missing files.

**Comment by Technoguyfication** [comment]
Isn't it possible to just create two separate backups to the same location, that way when the external HDD isn't plugged in it doesn't stop the main backup?

**Comment by ts678** [comment]
Using two backups also allows the backup for the maybe-there-maybe-not drive to test for drive presence and skip if not present.
[run-script-before](https://duplicati.readthedocs.io/en/latest/06-advanced-options/#run-script-before), see how it looks, give exit code:

https://github.com/duplicati/duplicati/blob/c7f9f8f5566b99d3189d9d32f24a888a8cdde231/Duplicati/Library/Modules/Builtin/run-script-example.sh#L19-L27
